pydatastructs/graphs/algorithms.py

```diff
@@ -993,3 +995,110 @@ def _job(graph: Graph, u: str):
     if len(L) != num_vertices:
         raise ValueError("Graph is not acyclic.")
     return L
+
+def lowest_common_ancestor(graph: Graph, vertex1: str, vertex2: str, algorithm: str) -> str:
+    """
+    Finds the lowest common ancestor of two vertices u and v of a directed acylic graph.
+    The LCA of two vertices u and v is defined as the vertex w which is an ancestor
+    of both u and v and is farthest from the root vertex.
+
+    Parameters
+    ==========
+
+    graph: Graph
+        The graph under consideration.
+    vertex1, vertex2: str
+        The names of the vertices in the graph whose lowest common
+        ancestor is to be found.
+    algorithm: str
+        The algorithm to be used.
+        Currently, following are supported,
+        'binary_lifting' -> Binary lifting algorithm as given in [1].
+
+    Returns
+    =======
+
+    str
+        The name of the vertex that is the lowest common ancestor of the two given
+        vertices in the given graph.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import Graph, AdjacencyListGraphNode, lowest_common_ancestor
+    >>> v_1 = AdjacencyListGraphNode('v_1')
+    >>> v_2 = AdjacencyListGraphNode('v_2')
+    >>> v_3 = AdjacencyListGraphNode('v_3')
+    >>> v_4 = AdjacencyListGraphNode('v_4')
+    >>> v_5 = AdjacencyListGraphNode('v_5')
+    >>> graph = Graph(v_1, v_2, v_3, v_4, v_5)
+    >>> graph.add_edge('v_1', 'v_2')
+    >>> graph.add_edge('v_1', 'v_3')
+    >>> graph.add_edge('v_3', 'v_4')
+    >>> graph.add_edge('v_3', 'v_5')
+    >>> lowest_common_ancestor(graph, 'v_2', 'v_5', 'binary_lifting')
+    'v_1'
+    >>> lowest_common_ancestor(graph, 'v_4', 'v_5', 'binary_lifting')
+    'v_3'
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Topological_sorting#Kahn's_algorithm
```

Comment:
Are binary lifting and Kahn's algorithm the same thing?

Reply:
No sorry I had to change the reference used. I'll do it right away.